frontend/src/lib/components/Annotations/AnnotationMarker.js

```diff
@@ -0,0 +1,133 @@
+                                                (data.created_by.first_name || data.created_by.email)) ||
+                                                name ||
+                                                email}
+                                        </b>
+                                        <i style={{ color: 'gray' }}>{humanFriendlyDetailedTime(data.created_at)}</i>
+                                    </div>
+                                    {(!data.created_by || data.created_by.id === id) && (
+                                        <DeleteOutlined
+                                            className="clickable"
+                                            onClick={() => {
+                                                onDelete(data.id)
+                                            }}
+                                        ></DeleteOutlined>
+                                    )}
+                                </Row>
+                                <span>{data.content}</span>
+                            </div>
+                        ))}
+                        {textAreaVisible && (
+                            <TextArea
+                                maxLength={300}
+                                style={{ marginBottom: 12 }}
+                                rows={4}
+                                value={textInput}
+                                onChange={e => setTextInput(e.target.value)}
+                            ></TextArea>
+                        )}
+                        {textAreaVisible ? (
+                            <Row justify="end">
+                                <Button
+                                    type="primary"
+                                    onClick={() => {
+                                        onCreate(textInput)
+                                        setTextInput('')
+                                        setTextAreaVisible(false)
+                                    }}
+                                >
+                                    Add
+                                </Button>
+                            </Row>
+                        ) : (
+                            <Row justify="end">
+                                <Button
+                                    type="primary"
+                                    onClick={() => {
+                                        setTextAreaVisible(true)
+                                    }}
+                                >
+                                    Add Annotation
+                                </Button>
+                            </Row>
+                        )}
+                    </div>
+                )
+            }
+            title={
+                <Row justify="space-between" align="middle">
+                    {label}
+                </Row>
+            }
+            {...{ ...(visible !== null && visible !== undefined && { visible }) }}
```

Comment:
If the visible prop is set at all the component becomes a controlled state where it won't automatically handle visibility state. I only want this in the case where I pass `visible` from above and in some cases I don't have to. That horrid logic is so that the visible prop is only _touched_ in specific cases

Reply:
Might be wise to add a `/* comment */` explaining this to future generations